lib/app/routes/account.rb

```diff
@@ -8,14 +8,16 @@ class Account < Core
         erb :"account/show", locals: { profile: Profile.new(current_user) }
       end
 
-      put '/account/email' do
+      put '/account' do
         if current_user.guest?
-          halt 403, "You must be logged in to edit your email settings"
+          halt 403, "You must be logged in to edit your account settings"
         end
 
-        current_user.email = params[:email]
-        current_user.save
-        flash[:success] = 'Updated email address.'
+        if current_user.update(params[:account])
+          flash[:success] = 'Updated account settings.'
+        else
+          flash[:error] = current_user.errors.full_messages.join(', ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.